### DISCUSSION TITLE
Details on R_SUSPICIOUS_IMAGES

### DISCUSSION THREAD

**Discussion by flaviozluca** [rspamd/rspamd#4277] [Q&A]
Friends!

Where can I find details on rules, such as R_SUSPICIOUS_IMAGES? It hit in the legit email, attached.
[Bem-vindo ao Programa Adere Saúde! - 'Sankyo' (no-reply@dsbr.com.br) - 2022-09-20 0745.zip](https://github.com/rspamd/rspamd/files/9608249/Bem-vindo.ao.Programa.Adere.Saude.-.Sankyo.no-reply%40dsbr.com.br.-.2022-09-20.0745.zip)

I didnt find a search in documentation.

Thanks!

**Comment by Omega3113** [comment]
It seems (!!!) that it provides a relation of text in the mail to the number of Images. Or specifically it does a wild assumption of the number of words that a image could contain and relates real words to total words.
So this should avoid Spam that messages the users without using words in the mail text.

**Comment by flaviozluca** [comment]
Thanks, Omega.

1) Is there a list of rspamd tests and a description of each I can see somewhere?

2) I see. Its basically the proportion of image to text, perhaps. Its something I miss in many emails. I wonder why, I even have other threads about this, see: https://github.com/rspamd/rspamd/discussions/3985

In this case, this legit emails has a few images, one followed by the next... there is no text you can read in the body of the message.

R_SUSPICIOUS_IMAGES(5) BAD_REP_POLICIES(0.1) BL_REPEAT(3) MIME_HTML_ONLY(0.2) R_EMPTY_IMAGE(2

EMPTY IMAGE also scored.

The topic seems to be complex. https://github.com/rspamd/rspamd/issues/525

IMHO, maybe it would be best to simply score when the proportion image to spam is big.

In this case, the 2 images tests scored up 7 points, seems a little too much.

And other spams that have single images don't get scored, unless something changed (see my other ticket I referred to).

Sorry to make things more confusing, it's complex to me. I don't even mind this FP, one shouldnt send single imaged emails anyway... but what disturbs me a bit is the ones that are not caught and understanding rules, although that a clear description of them might not be good so spammers would find out (doubt it, though).

Thanks.

**Comment by Omega3113** [comment]
To understand a score, i mainly look at the source code, which is hard for me, cause i never saw lua before and always need to look up rspamd internal functions.
The discussion you linked are a good example why spam filtering is hard. There is always someone who sends an obscure message, but as a mail you want to receive. My own mails to me would often get flagged for only containing a picture taken with my smartphone.
Looking up external images as requested would cause every tracking pixel to be activated. Most people would cry on this. 
If you find, that it doesn't work for you, adjust the scores.

**Comment by flaviozluca** [comment]
Thanks a lot.

If you get a chance, check my other post, strange, many false negatives with spams that have images. I believe there's space for improvement, although it might have improved. Will keep an eye.

All the best.

**Comment by flaviozluca** [comment]
Friend,

Check this one out, for example, a single image and no score regarding that. False negative.

Thanks.
[image spam.zip](https://github.com/rspamd/rspamd/files/9636587/image.spam.zip)
